File: src/main/java/com/dafttech/terra/engine/IDrawableInWorld.java
package com.dafttech.terra.engine;

import com.dafttech.terra.game.world.World;
import com.dafttech.terra.game.world.entities.Entity;

public interface IDrawableInWorld {
    public void update(World world, float delta);

    public void draw(Vector2 pos, World world, AbstractScreen screen, Entity pointOfView);
}

File: terra/src/main/scala/com/dafttech/terra/engine/IDrawableInWorld.scala
package com.dafttech.terra.engine

import com.dafttech.terra.game.world.World
import com.dafttech.terra.game.world.entities.Entity

trait IDrawableInWorld {
  def update(delta: Float)(implicit tilePosition: TilePosition)

  def draw(screen: AbstractScreen, pointOfView: Entity)(implicit tilePosition: TilePosition)
}